libcaf_core/caf/string_algorithms.cpp

```diff
@@ -94,11 +94,12 @@ void replace_all(std::string& str, std::string_view what,
   };
   auto i = next(str.begin());
   while (i != str.end()) {
-    auto before = static_cast<size_t>(std::distance(str.begin(), i));
-    str.replace(i, i + what.size(), with.begin(), with.end());
+    auto before = std::distance(str.begin(), i);
+    str.replace(i, i + static_cast<ptrdiff_t>(what.size()), with.begin(),
```

Comment:
We do this cast twice now, so I'd store the result of the cast in a variable.

Reply:
First static cast is `what.size()` and the second `with.size()` if that's what you meant?